CMakeLists.txt

```diff
@@ -79,6 +79,19 @@ SET(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}" CACHE STRING
     "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel Maintainer."
     FORCE)
 
+# Explicit request to use CCACHE
+if($ENV{USE_CCACHE})
```

Comment:
This is only being triggered if the env is explicitly set. Nevertheless [this](https://github.com/travis-ci/travis-ci/issues/5383#issuecomment-224630584) is the most common option I see around. But it will required activating sudo permissions. I'm not really sure the consequences of that.